app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -0,0 +1,51 @@
+module ForemanDeployments
+  module Tasks
+    class SearchTaskDefinition < BaseDefinition
+      class Action < BaseAction
+        def run
+          results = SearchTaskDefinition.create_object(input)
+          results = SerializableArray.new(results)
+          SearchTaskDefinition.create_output(results, output)
+        end
+      end
+
+      class SerializableArray < Array
+        def to_hash
+          {}
```

Comment:
We can call it a temporary workaround - basically it nullifies the `output_hash['results']` field in task's output, since it could not be serialized.